src/core/shape_inference/include/topk_shape_inference.hpp

```diff
@@ -39,19 +36,16 @@ void shape_infer(const TopK* op,
         auto normalized_axis = ov::normalize_axis(op, op->get_provided_axis(), input_rank, -input_rank, input_rank - 1);
         auto& dim_axis = output_shape[normalized_axis];
 
-        if (!is_dynamic_shape) {
-            std::vector<int64_t> k_val;
-            NODE_VALIDATION_CHECK(op,
-                                  get_data_as_int64<T>(1, op, k_val, constant_data),
-                                  "determined k is required to infer static shape");
-
+        std::vector<int64_t> k_as_vals;
+        bool k_is_set = get_data_as_int64<T>(1, op, k_as_vals, constant_data);
```

Comment:
this if would always work and the next `else if` would be dead code.
I propose to merge these two ifs into one and use `get_data_as_shape` template function from here: https://github.com/openvinotoolkit/openvino/blob/master/src/core/shape_inference/include/utils.hpp#L111

Reply:
As you suggested, I use `get_date_as_shape` template function to merge two ifs statements into one.